### PR TITLE
add a basic graph inspector tool to build_runner

### PR DESCRIPTION
**Pull request by jakemac53**
got a quick start on this - today its interactive but we might want to change that since you don't get much bash niceness but its definitely useful for inspecting graphs and provides a good extension point for future features.

Today this has two commands, `inspect` (for inspecting individual nodes), and `graph` (for inspecting the whole graph). We can rename these later too if we want.

### CODE REVIEW

File: build_runner/bin/graph_inspector.dart [hunk 1]
```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import 'dart:async';
```
Comment (natebosch) [inline build_runner/bin/graph_inspector.dart]: [nit] I'm also a fan of a blank line under the copyright before the first import

File: build_runner/bin/graph_inspector.dart [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        abbr: 'p', help: 'Filters nodes to a certain package');
+    argParser.addOption('pattern',
+        abbr: 'm', help: 'glob pattern for path matching');
+  }
+
+  @override
+  run() {
+    var showGenerated = argResults['generated'] as bool;
+    var showSources = argResults['original'] as bool;
+    Iterable<AssetNode> nodes;
```
Comment (natebosch) [inline build_runner/bin/graph_inspector.dart]: [optional] Do you ever use anything from the node other than `id`? If not we could make this an `Iterable<AssetId>` and some of the lines below could be shorted to `assetGraph.sources` and `assetGraph.outputs`.

File: build_runner/bin/graph_inspector.dart [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+      nodes = nodes.where((node) => glob.matches(node.id.path));
+    }
+
+    for (var node in nodes) {
+      _listAsset(node.id, stdout, indentation: '  ');
+    }
+  }
+}
+
+AssetId _getIdFromStringUri(String stringUri) {
```
Comment (natebosch) [inline build_runner/bin/graph_inspector.dart]: [nit] would prefer without `get`. `_idFromString` is clear to me.

File: build_runner/bin/graph_inspector.dart (hunk 1 above)
Comment (jakemac53) [reply build_runner/bin/graph_inspector.dart]: done

File: build_runner/bin/graph_inspector.dart (hunk 2 above)
Comment (jakemac53) [reply build_runner/bin/graph_inspector.dart]: done

File: build_runner/bin/graph_inspector.dart (hunk 3 above)
Comment (jakemac53) [reply build_runner/bin/graph_inspector.dart]: done

**Pull request merged**